lib/usdUfe/ufe/trf/XformOpUtils.cpp

```diff
@@ -116,51 +116,52 @@ std::vector<UsdGeomXformOp> getOrderedXformOps(const UsdPrim& prim)
 Ufe::Vector3d getTranslation(const Ufe::Matrix4d& m)
 {
     Ufe::Vector3d t;
-    getTRS(m, &t, nullptr, nullptr);
+    extractTRS(m, &t, nullptr, nullptr);
     return t;
 }
 
 Ufe::Vector3d getRotation(const Ufe::Matrix4d& m)
 {
     Ufe::Vector3d r;
-    getTRS(m, nullptr, &r, nullptr);
+    extractTRS(m, nullptr, &r, nullptr);
     return r;
 }
 
 Ufe::Vector3d getScale(const Ufe::Matrix4d& m)
 {
     Ufe::Vector3d s;
-    getTRS(m, nullptr, nullptr, &s);
+    extractTRS(m, nullptr, nullptr, &s);
     return s;
 }
 
+namespace internal {
+
 void getTRS(const Ufe::Matrix4d& m, Ufe::Vector3d* t, Ufe::Vector3d* r, Ufe::Vector3d* s)
 {
-    // Decompose new matrix to extract TRS.  Neither GfMatrix4d::Factor
-    // nor GfTransform decomposition provide results that match Maya,
-    // so use MTransformationMatrix.
-    MMatrix mm;
-    std::memcpy(mm[0], &m.matrix[0][0], sizeof(double) * 16);
-    MTransformationMatrix xformM(mm);
-    if (t) {
-        auto tv = xformM.getTranslation(MSpace::kTransform);
-        *t = Ufe::Vector3d(tv[0], tv[1], tv[2]);
+    // Decompose new matrix to extract TRS.
+    PXR_NS::GfMatrix4d usdMatrix;
+    std::memcpy(usdMatrix[0], &m.matrix[0][0], sizeof(double) * 16);
```

Comment:
This code is the same as the one from above. This one has the three output vars. Modified to use USD API instead of Maya API. Placed this into "internal" namespace as its not meant to be called directly. Instead you should always use the new DCC function "extractTRS" (which is overridable).